src/sysl/importers/import_swagger.py

```diff
@@ -180,17 +205,48 @@ def info_attrs(info, prefix=''):
                         qparams = dict()
 
                         if 'parameters' in body:
-                            qparams = [p for p in body['parameters']
-                                       if p.get('in') == 'query']
+                            qparams = []
+                            headerParams = []
+
+                            for key in body['parameters']:
+                                if key.get('in') == 'query':
+                                    qparams.append(key)
+                                if '$ref' in key:
+                                    headerParams.append(key['$ref'].rpartition('/')[2])
+
+                        if 'parameters' in swag:
+                            optionals = []
+                            requires = []
+
+                            for headerParam in headerParams:
+                                formattedParam = headerParam.replace("-", "_")
+                                if swag['parameters'][headerParam]['required']:
+                                    requires.append("\"{}\"".format(formattedParam))
+                                else:
+                                    optionals.append("\"{}\"".format(formattedParam))
+                                alias[formattedParam] = swag['parameters'][headerParam]['type']
 
-                        w(u'{}{}{}:',
+                        header = []
+                        requiredHeader = ''
+                        optionalHeader = ''
+
+                        if len(requires) > 0:
+                            requiredHeader = 'required_headers=[{}]'.format(', '.join(requires))
+                            header.append(requiredHeader)
+                        if len(optionals) > 0:
+                            optionalHeader = 'optional_headers=[{}]'.format(', '.join(optionals))
+                            header.append(optionalHeader)
+
+                        header = '[{}]'.format(','.join(header))
```

Comment:
if possible it would be good to factor this new code out as method that returns header and takes body, swag, etc as arguments. then it would be easier to understand this code and test it